### PR TITLE
US7355 Configuration

### PR DESCRIPTION
**Pull request by tcmacdonald**
This PR updates the crds-shared-header library so that you can include the bundle and instantiate the header manually. See the updated README file for details.

### CODE REVIEW

File: src/shared_header/index.ts [hunk 1]
```diff
@@ -5,50 +5,62 @@ import { AuthListener } from './auth_listener';
 export class SharedHeader {
 
   public container: any;
+  public options = {
+    el: '[data-header]',
+    contentBlockTitle: 'globalSharedHeader',
+    contentBlockCategories: ['common'],
+    contentBlockId: 564,
+    imgEndpoint: 'https://www.crossroads.net/',
```
Comment (seancdavis) [inline src/shared_header/index.ts]: `contentBlockTitle` and `imgEndpoint` are slightly different than what you mentioned in the README.

File: src/shared_header/index.ts [hunk 2]
```diff
@@ -5,50 +5,62 @@ import { AuthListener } from './auth_listener';
...
+  constructor(opts = {}) {
+    (<any>Object).assign(this.options, opts);
+    this._utils = new Utils(this.options);
   }
 
-  constructor() {}
-
   public render() {
+
+    // console.log(this.options.contentBlockTitle);
```
Comment (seancdavis) [inline src/shared_header/index.ts]: Can we kill this, or leaving it around intentionally?

File: src/shared_header/auth_listener.ts [hunk 3]
```diff
@@ -3,12 +3,17 @@ import { Utils } from '../utilities/_base';
 export class AuthListener {
 
   public _interval: any;
+  public _utils: Utils;
```
Comment (seancdavis) [inline src/shared_header/auth_listener.ts]: Out of curiosity, two questions:

1. Why the switch to using an instance and not the object directly? (Is it to be able to store the options once and reference them throughout?)

2. Why did you make `_utils` public in some cases and private in others?

File: src/index.html [hunk 4]
```diff
@@ -29,5 +29,31 @@
               onclick="document.cookie = 'userId=7708041'">Sign Nobody In</button>
     </div>
 
+    <% if (htmlWebpackPlugin.options.window) { %>
+    <script>
+      <% for (var varName in htmlWebpackPlugin.options.window) { %>
+        window['<%=varName%>'] = <%= JSON.stringify(htmlWebpackPlugin.options.window[varName]) %>;
```
Comment (seancdavis) [inline src/index.html]: Not to be super picky, but wondering why you didn't add spaces around `varName`? (Simply for my reference in knowing what you're thinking stylistically.)

File: src/shared_header/auth_listener.ts [hunk 5]
```diff
@@ -20,33 +25,33 @@ export class AuthListener {
...
       // Add the user's name to any [data-user-name] elements, which is stored
       // in a cookie, like the userId.
-      Utils.find('[data-user-name]').html(Utils.getCookie('username'));
+      this._utils.find('[data-user-name]').html(this._utils.getCookie('username'));
       // The user's avatar is available via the CMS, and is attached to any
       // [data-user-avatar] elements.
-      Utils.getUserAvatarUrl(Utils.getCookie('userId')).then(function(url) {
-        Utils.find('[data-user-avatar]').attr('src', url);
+      this._utils.getUserAvatarUrl(this._utils.getCookie('userId')).then(function(url) {
+        this._utils.find('[data-user-avatar]').attr('src', url);
```
Comment (seancdavis) [inline src/shared_header/auth_listener.ts]: `this` is `undefined` if the avatar url can not be retrieved. Perhaps a `try`/`catch` or similar?

File: config/webpack.common.js [hunk 6]
```diff
@@ -5,59 +5,43 @@ var HtmlWebpackPlugin = require('html-webpack-plugin');
...
-        context: './node_modules/crds-styles/assets/svgs/',
-        from: '*.svg',
-        to: 'assets/svgs'
-      }
-    ], { ignore: ['mock-data/*'] })
-  ]
+  plugins: plugins
 };
+
+module.exports = config;
```
Comment (seancdavis) [inline config/webpack.common.js]: I'm understanding all your changes but Webpack is still a bit mysterious to me. I'd love to have you walk me through this at some point, just so I understand what you changed. However, for the case of merging, I tested the build process and seems to be working as expected.

File: src/index.html [hunk 7]
```diff
@@ -29,5 +29,31 @@
...
+      <% } %>
+    </script>
+    <% } %>
+
+    <% for (var chunk in htmlWebpackPlugin.files.chunks) { %>
+    <script src="<%= htmlWebpackPlugin.files.chunks[chunk].entry %>"></script>
```
Comment (seancdavis) [inline src/index.html]: In this particular file, this line is loading `bundle.js` a second time. Chrome is smart enough to not waste energy making the second request, but wondering if you actually need this line for our `index.html` example file?

File: src/shared_header/auth_listener.ts (hunk 3 above)
Comment (tcmacdonald) [inline src/shared_header/auth_listener.ts]: 1. Yes
2. I was coding at midnight. Thanks for the catch!

File: src/shared_header/auth_listener.ts (hunk 5 above)
Comment (tcmacdonald) [inline src/shared_header/auth_listener.ts]: Resolved by hardcoding the avatar URL.

File: config/webpack.common.js (hunk 6 above)
Comment (tcmacdonald) [inline config/webpack.common.js]: Will do. Let me know when you get a chance.

**Pull request merged**